### PR TITLE
[WIP] Add carbon dioxide, cartridges and inflatable vest you can inflate and deflate

### PR DESCRIPTION
**Pull request by Salty-Panda**
#### Summary
Content "Life jacket you can inflate and deflate"

#### Purpose of change

I wanted to get a better hang on `"use_action"`

#### Describe the solution

It's TOOL_ARMOR, with MAGAZINE_WELL which stores CO2 cartridges.
When activated it consumes a single charge of CO2 from inside the cartridge and transforms into an inflated version.
__To make this part work I had to make a bugfix, #47870. Until it's merged this item doesn't consume any charges, like a cellphone.
It also requires #47887 to be fixed.__
When you activate the inflated version you can freely deflate it, without consuming any charges.
You can freely unload and reload it with a cartridge.

Spawning: TBD

#### Describe alternatives you've considered

Reading the code or something

#### Testing

Activated, deactivated, played around, and debugged a lot during making the bugfixing part.

### CODE REVIEW

File: data/json/items/ammo.json [hunk 1]
```diff
@@ -1141,6 +1141,23 @@
...
+    "id": "co2",
+    "category": "chems",
+    "price": 300,
+    "price_postapoc": 25,
+    "name": { "str_sp": "carbon dioxide" },
+    "symbol": "=",
+    "color": "white",
+    "container": "bottle_plastic",
+    "sealed": false,
+    "description": "Compressed carbon dioxide used in inflatable devices",
```
Comment (actual-nh) [inline data/json/items/ammo.json:1155]:
```suggestion
    "description": "Compressed carbon dioxide, usually used in inflatable devices",
```

File: data/json/items/armor/torso_clothes.json [hunk 2]
```diff
@@ -174,6 +174,80 @@
...
+    "material_thickness": 1,
+    "charges_per_use": 1,
+    "use_action": {
+      "type": "transform",
+      "target": "flotation_vest_inflatable_on",
+      "menu_text": "pull the cord",
+      "msg": "You pull the cord.",
+      "need_charges": 1,
+      "need_worn": true,
+      "need_charges_msg": "There is nothing to inflate th vest with.",
```
Comment (actual-nh) [inline data/json/items/armor/torso_clothes.json:208]:
```suggestion
      "need_charges_msg": "There is nothing to inflate the vest with.",
```

File: data/json/items/armor/torso_clothes.json [hunk 3]
```diff
@@ -174,6 +174,80 @@
     "material_thickness": 4,
     "flags": [ "FLOTATION", "WATER_FRIENDLY", "BELTED" ]
   },
+  {
+    "id": "flotation_vest_inflatable",
+    "repairs_like": "flotation_vest_ms",
+    "type": "TOOL_ARMOR",
+    "name": { "str": "inflatable flotation vest" },
+    "description": "A brightly-colored device designed to keep you upright and floating in water.  Activate to inflate",
```
Comment (actual-nh) [inline data/json/items/armor/torso_clothes.json:182]:
```suggestion
    "description": "A brightly-colored device designed to keep you upright and floating in water.  Activate to inflate.",
```

File: data/json/items/magazine/gas.json [hunk 4]
```diff
@@ -36,5 +36,24 @@
     "count": 240,
     "flags": [ "NO_UNLOAD", "NO_RELOAD" ],
     "pocket_data": [ { "pocket_type": "MAGAZINE", "rigid": true, "ammo_restriction": { "weldgas": 240 }, "watertight": true } ]
+  },
+  {
+    "id": "co2_cartridge",
+    "looks_like": "pressurized_tank",
+    "type": "MAGAZINE",
+    "name": { "str": "carbon dioxide cartridge" },
+    "description": "A small aluminum cartridge designed for holding pressurized carbon dioxide.",
```
Comment (actual-nh) [inline data/json/items/magazine/gas.json:45]:
```suggestion
    "description": "A small aluminum cartridge designed to hold pressurized carbon dioxide.",
```

File: data/json/items/magazine/gas.json [hunk 5]
```diff
@@ -36,5 +36,24 @@
...
+    "volume": "260 ml",
+    "price": 600,
+    "price_postapoc": 100,
+    "material": [ "aluminum" ],
+    "symbol": "#",
+    "color": "light_gray",
+    "ammo_type": [ "co2" ],
+    "capacity": 1,
+    "count": 1,
+    "flags": [ "NO_UNLOAD", "NO_RELOAD" ],
```
Comment (actual-nh) [inline data/json/items/magazine/gas.json:56]: At least without a compressor (and means of generating or separating CO2).

**Review by actual-nh — COMMENTED** [review]
Useful!

File: data/json/items/magazine/gas.json (hunk 5 above)
Comment (Salty-Panda) [reply data/json/items/magazine/gas.json:56]: oxyacetylene is craftable, but all containers that store it have both of those flags too.

File: data/json/items/ammo.json [hunk 6]
```diff
@@ -1141,6 +1141,23 @@
...
+  {
+    "type": "AMMO",
+    "id": "co2",
+    "category": "chems",
+    "price": 300,
+    "price_postapoc": 25,
+    "name": { "str_sp": "carbon dioxide" },
+    "symbol": "=",
+    "color": "white",
+    "container": "bottle_plastic",
```
Comment (Qrox) [inline data/json/items/ammo.json:1153]:
```suggestion
    "container": "co2_cartridge",
```

File: data/json/items/ammo.json [hunk 7]
```diff
@@ -1141,6 +1141,23 @@
...
+    "id": "co2",
+    "category": "chems",
+    "price": 300,
+    "price_postapoc": 25,
+    "name": { "str_sp": "carbon dioxide" },
+    "symbol": "=",
+    "color": "white",
+    "container": "co2_cartridge",
+    "sealed": false,
+    "description": "Compressed carbon dioxide, usually used in inflatable devices",
```
Comment (chaohedgehogs) [inline data/json/items/ammo.json:1155]:
```suggestion
    "description": "Compressed carbon dioxide, usually used in inflatable devices.",
```

**Comment by I-am-Erk** [comment]
Closing as stale. Please feel free to reopen when you're back around.

**Pull request closed without merging**